### ISSUE TITLE
`object-curly-spacing`: should ignore TypeScript index signatures

### ISSUE THREAD

**Issue by nwhittaker** [eslint-stylistic/eslint-stylistic#767]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

Given an `object-curly-spacing` rule configured as `always` with `arraysInObjects` disabled:
```
'@stylistic/object-curly-spacing': ['error', 'always', { arraysInObjects: false }],
```

And a TypeScript type with an index signatures:
```ts
type Obj = { [index: string]: unknown }
```

ESLint treats the index signature as an array and reports an error:
```
error  There should be no space after '{'  @stylistic/object-curly-spacing
```

### Reproduction

https://stackblitz.com/edit/github-bf7hmxtz

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by keijokapp** [comment]
This behavior is from typescript-eslint/typescript-eslint#2892 and typescript-eslint/typescript-eslint#3176. Looking at the [source code](https://github.com/eslint-stylistic/eslint-stylistic/blob/753ee3e25cf7d1d154fa18866ad28c3fb7bdeb2d/packages/eslint-plugin-ts/rules/object-curly-spacing/object-curly-spacing.ts#L178-L182) and the [test cases](https://github.com/eslint-stylistic/eslint-stylistic/blob/753ee3e25cf7d1d154fa18866ad28c3fb7bdeb2d/packages/eslint-plugin-ts/rules/object-curly-spacing/object-curly-spacing.test.ts#L719-L739) here (introduced by 753ee3e2 in v0.0.3), this appears to be a deliberate behavior.

The reasoning behind it is not explained. For context, computed properties (which look a lot like mapped types and indexed signatures), do not appear to have such special handling.
